### ISSUE TITLE
Chef Workstation 2020 Testing

### ISSUE THREAD

**Issue by btm** [chef/chef-workstation#1167]
Chef Workstation 20.05 needs additional testing before release. Related: #1059

Get the latest build from current and install it: `mixlib-install download chef-workstation -c current`. Must be at least 20.5.26+.

Ensure all commands run are only coming from the chef workstation install. No other ruby installations, rubygems, or other chef products should be in the PATH.

Test appropriate functionality, leave a comment on this issue regarding the results including noting all/any ruby warnings.

If everything tested works, and there are no Ruby deprecation errors to clean up, check the checkbox below.

- [x] inspec # the built in one
- [x] test-kitchen with inspec
- [x] cookstyle

### Knife Plugins
Ensure basic functionality of each plugin works
- [x] knife acl, e.g. knife acl show nodes NODE_NAME
- [x] knife tidy, e.g. knife tidy server report
- [x] knife opc, e.g. knife opc org create
- [x] knife windows, e.g. knife wsman test
- [x] knife bootstrap (ssh, winrm, cross-platforms)

### Clouds and virtualization
Ensure we can use the knife plugin to create an instance, and the test-kitchen driver works if available.

- [x] azure (test-kitchen with azure-rm and knife azure)
- [x] ec2 (test-kitchen with ec2 and knife ec2)
- [x] google (test-kitchen with google and knife google)
- [x] test-kitchen with hyperv
- [x] test-kitchen with vagrant

### Additional clouds/virtualization
These clouds are lower priority as they may be difficult to get access to an instance of them

- [x] test-kitchen with digital ocean
- [ ] test-kitchen with openstack
- [x] vcenter (test-kitchen with vcenter and knife-vcenter)
- [x] knife-vsphere

### Additional tests
- [x] knife push - requires a push jobs server.
- [x] stove, e.g. stove login # logs in to supermarket
- [x] berkshelf, e.g. berks install
- [x] https://github.com/chefspec/chefspec/issues/986

**Comment by dheerajd-msys** [comment]
**Chef Workstation Setup**
===============================
```
1. We need to remove installed chef/chefdk package from our machine

2. check status of chef/chefdk
   dpkg --status chef/chefdk

3. Remove installed package
   sudo dpkg -P chef/chefdk

4. Download chef-worksation for your ubuntu machine from current channel
   https://downloads.chef.io/chef-workstation/current

5. Install chef-workstaion
    sudo dpkg -i <downloaded-chef-workstation>

6. Set your system PATH for chef-workstation

dheerajdubey@PUN-LAP-DHEERAJ:~$ echo $PATH
/home/dheerajdubey/.rvm/gems/ruby-2.6.0/bin:/home/dheerajdubey/.rvm/gems/ruby-2.6.0@global/bin:/usr/share/rvm/rubies/ruby-2.6.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/share/rvm/bin

dheerajdubey@PUN-LAP-DHEERAJ:~$ export PATH=/opt/chef-workstation/embedded/bin/:$PATH

7. Set your GEM_PATH & GEM_HOME to /opt/chef-workstation/embedded/lib/ruby/gems/2.7.0
   export GEM_HOME=/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0
   export GEM_PATH=/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0

8. Validate using :
   gem env

9. By this time you should be able to see "knife -v" output 

10. dheerajdubey@PUN-LAP-DHEERAJ:~/.aws$ chef-cli -v
Chef Workstation version: 20.5.27
Chef Infra Client version: 16.0.287
Chef InSpec version: 4.18.111
Chef CLI version: 2.0.10
Test Kitchen version: 2.5.0
Cookstyle version: 6.3.4

11.
dheerajdubey@PUN-LAP-DHEERAJ:~/.aws$ which ruby
/opt/chef-workstation/embedded/bin/ruby
```

**Comment by dheerajd-msys** [comment]
```
dheerajdubey@PUN-LAP-DHEERAJ:~/.aws$ knife ec2 server create -I ami-xxxxx -U ec2-user -i ddubey-msys-chef.pem -N ddubey-amazon-msys --sudo --use-sudo-password -c knife.rb

WARN: Deprecation CHEF-29 from /opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/knife-ec2-1.0.36/lib/chef/knife/helpers/ec2_base.rb:183:in `locate_config_value'

  Use of config_value without a fallback_key is deprecated.  Knife plugin authors should access the config hash directly, which does correct merging of cli and config options.

Please see https://docs.chef.io/deprecations_knife_bootstrap_apis/ for further details and information on how to correct this problem.
WARN: Deprecation CHEF-29 from /opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/knife-ec2-1.0.36/lib/chef/knife/helpers/ec2_base.rb:183:in `locate_config_value'
```

**Comment by lamont-granquist** [comment]
@dheerajd-msys I'm working on fixing all of that in knife-ec2 right now.  I've fixed knife-azure already and master of that could be Q/A tested.  All the rest of the knife plugins need to be fixed, but we should probably coordinate if we're going to do that in parallel.

**Comment by dheerajd-msys** [comment]
Reports are getting generated only in JSON format for `knife tidy` https://github.com/chef/chef-workstation/issues/1178#issuecomment-629220629

**Comment by dheerajd-msys** [comment]
**Fail**(IN cases when the user has no read permission to any of the organization) https://github.com/chef/chef-workstation/issues/1178#issuecomment-629226142 and works well  when we provide `--orgs` option for `knife tidy server report` like https://github.com/chef/chef-workstation/issues/1178#issuecomment-629218022

**Comment by btm** [comment]
@dheerajd-msys I commented on https://github.com/chef/chef-workstation/issues/1178#issuecomment-629267514 that those issues in knife-tidy are fine. We're just looking for regressions, critical failures, and deprecation warnings here.

**Comment by dheerajd-msys** [comment]
Thank you @btm.

**Comment by tyler-ball** [comment]
chefspec is not tracked in this board on Zenhub so I added https://github.com/chefspec/chefspec/issues/986 in the card description

**Comment by dheerajd-msys** [comment]
@btm https://github.com/chefspec/chefspec/issues/986 is fixed via https://github.com/chefspec/chefspec/pull/988

**Comment by dheerajd-msys** [comment]
chef/chef#9922 is fixed in chef/chef#10298 and available in `Chef Infra Client v16.4.35+` and `Chef workstation v20.8.125`
Tested with both successfully.
